File: Develop/src/BankAccount.java
public class BankAccount {
    private long accountNumber;
    private double amount;

    public char getType() {
        return type;
    }

    private char type;

    public BankAccount(long accountNumber,char type) {
        this.type= type;
        this.accountNumber = accountNumber;


    }

    public long getAccountNumber() {
        return accountNumber;

    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;

    }

    public void agregarDinero(double amount){
        if(type=='A'){agregarDineroA(amount);}
        if(type=='B'){agregarDineroB(amount);}
        if (type=='C'){this.amount=this.amount+amount;
            System.out.println("Su deposito fue EXITOSO");}
    }

    private  void agregarDineroA(double amount){
        if (this.amount+amount<=50000){
            this.amount+=amount;
            System.out.println("Su deposito fue EXITOSO");
        }else{
            System.out.println("Su amount excede la cantidad maxima");
            this.amount=this.amount;
        }
    }
    private  void agregarDineroB(double amount){
        if (this.amount+amount<=100000){
            this.amount+=amount;
            System.out.println("Su deposito fue EXITOSO");
        }else{
            System.out.println("Su amount excede la cantidad maxima");
            this.amount=this.amount;
        }

    }
    public void quitarDinero(double amount){
        if(type=='A'){sacarDineroA(amount);}
        if(type=='B'){sacarDineroB(amount);}
        if (type=='C'){ if (this.amount-amount>=0){
            this.amount-=amount;
            System.out.println("Su retiro fue EXITOSO");
        }else{
            System.out.println("No tiene suficiente dinero");

        }
        }
    }

    private void sacarDineroA(double amount){
        if (this.amount-amount>=0){
            this.amount-=amount;
            System.out.println("Su retiro fue EXITOSO");
        }else{
            System.out.println("No tiene suficiente dinero");

        }
    }
    private void sacarDineroB(double amount){
        if (this.amount-amount>=0){
            this.amount-=amount;
            System.out.println("Su retiro fue EXITOSO");
        }else{
            System.out.println("No tiene suficiente dinero");

        }
    }
}

File: Develop/BankAccount.py
class BankAccount:
    def __init__(self, account_number, type):
        self.account_number = account_number
        self.amount = 0
        self.type = type

    def get_account_number(self):
        return self.account_number

    def get_amount(self):
        return self.amount

    def set_amount(self, amount):
        self.amount = amount

    def agregar_dinero(self, amount):
        if self.type == 'A':
            self.agregar_dinero_a(amount)
        elif self.type == 'B':
            self.agregar_dinero_b(amount)
        elif self.type == 'C':
            self.amount += amount
            print("Su deposito fue EXITOSO")

    def agregar_dinero_a(self, amount):
        if self.amount + amount <= 50000:
            self.amount += amount
            print("Su deposito fue EXITOSO")
        else:
            print("Su monto excede la cantidad máxima")
            self.amount = self.amount

    def agregar_dinero_b(self, amount):
        if self.amount + amount <= 100000:
            self.amount += amount
            print("Su deposito fue EXITOSO")
        else:
            print("Su monto excede la cantidad máxima")
            self.amount = self.amount

    def quitar_dinero(self, amount):
        if self.type == 'A':
            self.sacar_dinero_a(amount)
        elif self.type == 'B':
            self.sacar_dinero_b(amount)
        elif self.type == 'C':
            if self.amount - amount >= 0:
                self.amount -= amount
                print("Su retiro fue EXITOSO")
            else:
                print("No tiene suficiente dinero")

    def sacar_dinero_a(self, amount):
        if self.amount - amount >= 0:
            self.amount -= amount
            print("Su retiro fue EXITOSO")
        else:
            print("No tiene suficiente dinero")

    def sacar_dinero_b(self, amount):
        if self.amount - amount >= 0:
            self.amount -= amount
            print("Su retiro fue EXITOSO")
        else:
            print("No tiene suficiente dinero")
